src/wazuh_db/wdb_integrity.c

```diff
@@ -469,3 +470,41 @@ int wdbi_query_clear(wdb_t * wdb, wdb_component_t component, const char * payloa
 
     return ret_val;
  }
+
+int wdbi_check_sync_status (wdb_t *wdb, wdb_component_t component) {
+    cJSON* j_sync_info = NULL;
+    int result = 1;
+
+    sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_SYNC_GET_INFO);
+
+    if (stmt == NULL) {
+        return OS_INVALID;
+    }
+
+    sqlite3_bind_text(stmt, 1, COMPONENT_NAMES[component], -1, NULL);
+    j_sync_info = wdb_exec_stmt(stmt);
+
+    if (!j_sync_info) {
+        mdebug1("wdb_exec_stmt(): %s", sqlite3_errmsg(wdb->db));
+        return OS_INVALID;
+    }
+
+    cJSON* j_last_attempt = cJSON_GetObjectItem(j_sync_info->child, "last_attempt");
+    cJSON* j_last_completion = cJSON_GetObjectItem(j_sync_info->child, "last_completion");
+
+    if (!j_last_attempt || !j_last_completion) {
+        mdebug1("Failed to get agent's sync status data");
+        cJSON_Delete(j_sync_info);
```

Comment:
We can avoid this extra cJSON_Delete by handling the following inside an else statement, like:

```
int result = 0;
if (cJSON_IsNumber(j_last_attempt) && cJSON_IsNumber(j_last_completion)) {
    int last_attempt = j_last_attempt->valueint;
    int last_completion = j_last_completion->valueint;
    if (last_completion != 0 && last_completion >= last_attempt ) {
        result =1;
    }
}
else {
    mdebug1("Failed to get agent's sync status data");
    result = OS_INVALID;
}
cJSON_Delete(j_sync_info);
return result;
```

Reply:
The method was refactored.